eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -124,35 +124,20 @@ jobs:
             Pool: $(Pool)
             ${{ insert }}: ${{ parameters.EnvVars }}
 
-      - ${{ if parameters.TestResourceDirectories }}:
-        - ${{ each directory in parameters.TestResourceDirectories }}:
-          - template: /eng/common/TestResources/deploy-test-resources.yml
-            parameters:
-              Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
-              ServiceDirectory: '${{ directory }}'
-              SubscriptionConfiguration: $(SubscriptionConfiguration)
-              ArmTemplateParameters: $(ArmTemplateParameters)
-              UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
-              ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
-              SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
-              EnvVars:
-                Pool: $(Pool)
-                ${{ insert }}: ${{ parameters.EnvVars }}
-      - ${{ if not(parameters.TestResourceDirectories) }}:
-          - template: /eng/common/TestResources/deploy-test-resources.yml
-            parameters:
-              Location: ${{ coalesce(parameters.Location, parameters.CloudConfig.Location) }}
-              ServiceDirectory: '${{ parameters.ServiceDirectory }}'
-              SubscriptionConfiguration: $(SubscriptionConfiguration)
-              ArmTemplateParameters: $(ArmTemplateParameters)
-              UseFederatedAuth: ${{ parameters.UseFederatedAuth }}
-              ServiceConnection: ${{ parameters.CloudConfig.ServiceConnection }}
-              SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
-              EnvVars:
-                Pool: $(Pool)
-                ${{ insert }}: ${{ parameters.EnvVars }}
+      - ${{ each directory in coalesce(parameters.TestResourceDirectories, split(parameters.ServiceDirectory, '|')) }}:
```

Comment:
Are you enabling a new custom way of defining multi service directory with `|` here?

Reply:
Or is this just a way to cast to array.

Reply:
No the `|` here could be anything not already in the string. It is essentially just allowing me to convert a string into an array with only one element so we can pick one list or the other.

Reply:
Maybe it isn't documented but I didn't find that https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops